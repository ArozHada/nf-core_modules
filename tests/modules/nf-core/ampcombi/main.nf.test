nextflow_process {

    name "Test Process AMPCOMBI"
    script "modules/nf-core/ampcombi/main.nf"
    process "AMPCOMBI"
    config "tests/modules/nf-core/ampcombi/nextflow.config"
    tag "ampcombi"

    test("Should run without failures") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [[id:'sample_1'],
                            [file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/ampcombi/test_files/ampir/sample_1/sample_1.ampir.tsv', checkIfExists: true),
                            file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/ampcombi/test_files/amplify/sample_1/sample_1_amplify.tsv', checkIfExists: true)]
                           ]
                input[1] = file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/ampcombi/test_faa/sample_1.faa', checkIfExists: true)
                
                input[2] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match() }
            )
        }

    }

}
