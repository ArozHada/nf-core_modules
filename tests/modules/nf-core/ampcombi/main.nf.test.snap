{
    "Should run without failures": {
        "content": [
            [
                "ampcombi.log:md5,4611addfa6fb326b554e055777ef0355",
                [
                    "sample_1_amp.faa:md5,9d7f55164e093c8eabbe84ceec8135de",
                    "sample_1_ampcombi.csv:md5,e550e389617a1139387310544c6e555f",
                    "sample_1_diamond_matches.txt:md5,2ab4e98a261f400d8729ef477b3e3c64"
                ],
                "versions.yml:md5,b9ec8aabc26aa6c5129da76c119e60b0"
            ]
        ],
        "timestamp": "2023-04-06T00:34:23+0000"
    }
}